### PR TITLE
Fixes AI's being unable to call shuttle using comm console.

### PR DESCRIPTION
**Pull request by AnturK**
One day .machine will be gone and i'll shed a tear.

### CODE REVIEW

**Comment by YakumoChen** [comment]
I would strain to question why this is strictly needed since the AI can do it without the console but alas

File: code/game/objects/objs.dm [hunk 1]
```diff
@@ -138,19 +138,23 @@
...
-				is_in_use = 1
-				src.interact(M)
-		var/ai_in_use = AutoUpdateAI(src)
-
-		if(!ai_in_use && !is_in_use)
-			obj_flags &= ~IN_USE
+		var/is_in_use = FALSE
+		if(update_viewers)
+			var/list/nearby = viewers(1, src)
+			for(var/mob/M in nearby)
```
Comment (SpaceManiac) [inline code/game/objects/objs.dm]: IIRC, `viewers()` gets faster if you don't save it out to a list, as in `for(var/mob/M in viewers(1, src))`

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
Breaks the ability of a human to call the shuttle

File: code/game/machinery/computer/communications.dm [hunk 2]
```diff
@@ -45,8 +45,9 @@
 
 /obj/machinery/computer/communications/process()
 	if(..())
-		if(state != STATE_STATUSDISPLAY && state != STATE_CALLSHUTTLE && state != STATE_PURCHASE && state != STATE_VIEWMESSAGE)
-			updateDialog()
+		var/skip_ai = aistate != STATE_STATUSDISPLAY && aistate != STATE_CALLSHUTTLE && aistate != STATE_PURCHASE && aistate != STATE_VIEWMESSAGE
+		var/skip_machine_users = state != STATE_STATUSDISPLAY && state != STATE_CALLSHUTTLE && state != STATE_PURCHASE && state != STATE_VIEWMESSAGE
+		updateDialog(!skip_machine_users,!skip_ai)
```
Comment (SpaceManiac) [inline code/game/machinery/computer/communications.dm]: I think these checks are just flipped (variables named wrong)

**Comment by moonheart08** [comment]
@Anonmare This. So much this. I fight the constantly updating UI a *lot*. I now just flat out close the other UI when making an announcement.

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**